torch/nn/modules/transformer.py

```diff
@@ -132,6 +134,28 @@ def forward(self, src: Tensor, tgt: Tensor, src_mask: Optional[Tensor] = None, t
             src_key_padding_mask: the Tensor mask for src keys per batch (optional).
             tgt_key_padding_mask: the Tensor mask for tgt keys per batch (optional).
             memory_key_padding_mask: the Tensor mask for memory keys per batch (optional).
+            src_is_causal: If specified, applies a causal mask as tgt mask.
+                Default: ``None``; try to detect a causal mask.
+                Warning:
+                ``src_is_causal`` provides a hint that ``src_mask`` is
+                the causal mask. Providing incorrect hints can result in
+                incorrect execution, including forward and backward
+                compatibility.
+            tgt_is_causal: If specified, applies a causal mask as tgt mask.
+                Default: ``None``; try to detect a causal mask.
+                Warning:
+                ``tgt_is_causal`` provides a hint that ``tgt_mask`` is
+                the causal mask. Providing incorrect hints can result in
+                incorrect execution, including forward and backward
+                compatibility.
+            memory_is_causal: If specified, applies a causal mask as
+                memory mask.
```

Comment:
My opinion was that cross-attention is almost never going to be causal, so it would be a waste to make the check most of the time. In contrast, the encoder self-attention may still be causal when the `TransformerEncoder` is used for a decoder-only model.

Reply:
Are you arguing it should not even exist as a variable?  (We currently have it here, and for decoder, but only True/False, no auto detection (None) to avoid the overhead of checking, and init to `Flase`, i.e., nobody has to bother.

The question is which one is less surprising, having it (because it's not used frequently) or not having it (for consistency of _mask with _is_causal).